bundles/org.openhab.binding.tibber/src/main/java/org/openhab/binding/tibber/internal/handler/TibberHandler.java

```diff
@@ -297,7 +297,31 @@ public void updateRequest() throws IOException {
         getURLInput(BASE_URL);
         if ("true".equals(rtEnabled) && !isConnected()) {
             logger.debug("Attempting to reopen Websocket connection");
-            open();
+
+            TibberPriceConsumptionHandler tibberQuery = new TibberPriceConsumptionHandler();
+            InputStream wsURL = tibberQuery.getWebsocketUrl();
+            String wsResponse = HttpUtil.executeUrl("POST", BASE_URL, httpHeader, wsURL, null, REQUEST_TIMEOUT);
+
+            JsonObject wsobject = (JsonObject) JsonParser.parseString(wsResponse);
+            JsonObject dataObject = wsobject.getAsJsonObject("data");
+            if (dataObject != null) {
+                JsonObject viewerObject = dataObject.getAsJsonObject("viewer");
+                if (viewerObject != null) {
+                    JsonElement subscriptionElement = viewerObject.get("websocketSubscriptionUrl");
+                    if (subscriptionElement != null) {
+                        subscriptionURL = subscriptionElement.toString().replaceAll("^\"|\"$", "");
+                    }
+                }
+            }
+            String url = subscriptionURL;
```

Comment:
This code is duplicated from line 150-164. Could it be extracted to a private method?

Reply:
The common part of requesting subscription url is now combined into a common method.